domain/schema/controller.go

```diff
@@ -664,6 +665,30 @@ CREATE TABLE model_migration_minion_sync (
 );`)
 }
 
+func modelLastLogin() schema.Patch {
+	return schema.MakePatch(`
+CREATE TABLE model_last_login (
+    model_uuid TEXT NOT NULL,
+    user_uuid TEXT NOT NULL,
+    time TIMESTAMP,
```

Comment:
Should this be NOT NULL?

Reply:
Good point, added a not null.

I had left it as nullable because ideally we would actually insert a row in here on user creation (with a null timestamp) and use a update to add last logins. As it is we use an upsert to add the row in. I discussed this with Joe and right now we don't have a centralised place where we add users so we're not really ready to switch to doing that.